emcc.py

```diff
@@ -1385,6 +1385,10 @@ def is_supported_link_flag(f):
         # include it by default.  Should be eliminated by meta-DCE if unused.
         shared.Settings.EXPORTED_FUNCTIONS += ['_setThrew']
 
+    if shared.Settings.WASM_BACKEND:
+      if '_main' in shared.Settings.EXPORTED_FUNCTIONS:
+        shared.Settings.EXPORTED_FUNCTIONS += ['___main_argc_argv']
```

Comment:
doesn't this need to depend on whether we are in standalone mode?

this adds an export - do we need to export both `_main` and `__main_argc_argv`? (if one calls the other, we just need the outer one I think)

Reply:
All this does right now export it *if* it exists.  Neither symbols need to actually exits, it just stops them from being DCE'd by the linker if they do exist.

Reply:
Oh right, we only error on exports the user provides, not ones we add internally, thanks.